lib/queue.js

```diff
@@ -619,6 +619,8 @@ Queue.prototype.processJob = function(job, renew){
         renew = true;
         lockRenewId = _this.timers.set('lockRenewer', _this.LOCK_RENEW_TIME / 2, lockRenewer);
       }
+      // TODO: if we failed to re-acquire the lock while trying to renew, should we let the job
+      // handler know and cancel the timer?
       return locked;
```

Comment:
Ideally we tell the job processor somehow that the job is now aborted. It's not really a "failure" in the bull sense because the job is still be worked on by the second worker. Perhaps we introduce a new "aborted" event and ask job processors to listen to it and abort their processing if one of their jobs gets stolen. Though if a job gets stolen, then it means that processor is behind in its event loop, so it's probably not going to get that event anyways.